src/Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native/SoftwareEntitlementClient.h

```diff
@@ -52,7 +52,7 @@ class Entitlement
 // entitlement validation failure.
 //
 std::unique_ptr<Entitlement> GetEntitlement(
-    const std::string& url,
+    std::string& url,
```

Comment:
Fab, I would definitely prefer to remove the string pass-by reference and make a copy. It would keep the purpose of the method to only getting the entitlement and prevent the url from being changed when coming back from the method - encapsulation of purpose is preferable.